backend/app/services/adapters/bot_kinds.py

```
@@ -329,53 +344,32 @@ def create_with_user(
-                "namespace": "default",
-                "labels": {"type": shell_type},
-            },
-            "status": {"state": "Available"},
-            "apiVersion": "agent.wecode.io/v1",
-        }
+        # Get shell info by name (resolves actual shell_type from shell_name)
+        # The shell_name is the name of the user-selected Shell (custom or public)
+        try:
+            shell_info = get_shell_info_by_name(db, obj_in.shell_name, user_id)
+        except ValueError as e:
+            raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.6)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)